src/mlmc/quantity.py

```diff
@@ -112,31 +112,34 @@ def size(self) -> int:
         """
         return self.qtype.size()
 
-    def get_cache_key(self, chunk_spec):
+    def get_cache_key(self, level_id=None, chunk_id=None, chunk_slice=None, n_samples=None):
         """
         Create cache key
-        :param chunk_spec: ChunkSpec instance
-        :return: tuple
         """
-        return (chunk_spec.level_id, chunk_spec.chunk_id, chunk_spec.n_samples, chunk_spec.chunk_size, id(self))  # redundant parentheses needed due to py36, py37
+        chunk_size = None
+        if chunk_slice is not None:
```

Comment:
Why can be `chunk_slice` None?

Reply:
`Quantity._result_qtype` calls samples method directly:

`chunks_quantity_level = [q.samples(level_id=0, chunk_id=0, n_samples=10) for q in quantities]`

There is no prior information from storage.